### PR TITLE
[release/2.1] Include transitive refs in meta-package .nuspec files

### PR DESCRIPTION
**Pull request by dougbu**
- #30279
- ensure dependencies are up-to-date even when PatchConfig.props is empty
- also includes https://github.com/dotnet/aspnetcore/pull/30164/commits/5bafa2d364e6fd3e1bddfca0034b9a81b4e14db7 from #30164

### CODE REVIEW

**Comment by Pilchie** [comment]
Does this mean we'll publish a new version of the meta-packages for all patches, with their dependencies flattened?

I'm not against that (it makes things much better for people using 2.1 on .NET Framework for instance, but in the past there were some concerns about needing to update templates, and that affecting the offline package cache size in the SDK, etc.

**Comment by dougbu** [comment]
@Pilchie our 2.1 builds always produce the two meta-packages. This will flatten only flatten dependencies from dotnet/CoreFx and dotnet/core-setup once it's ready for review. Basically, it'll be similar to the 3.1 and 5.0 special cases we have for some dotnet/runtime repos e.g. https://github.com/dotnet/aspnetcore/blob/release/5.0/eng/Versions.props#L173-L186

For now, this PR is
1. Not inclusive enough because it doesn't cover a few dotnet/CoreFx and dotnet/core-setup dependencies that are not yet referenced directly anywhere
2. Too inclusive because I updated a bunch of third-party dependencies that land in our meta-packages but are highly unlikely to get new versions we worry about. My preference here is to change their handling only if we need to react to security issues in those dependencies.

**Comment by dougbu** [comment]
Real test will be https://dev.azure.com/dnceng/internal/_build/results?buildId=1026812

**Comment by dougbu** [comment]
Waiting for #30729 to work. Looks as if only ProdCon builds work internally for release/2.1☹️

**Comment by dougbu** [comment]
Do **not** merge pro-actively. Will rebase on #30729 commits to enable internal testing and validate that.

**Comment by dougbu** [comment]
Confirmed this PR does what it says on the tin by comparing https://dev.azure.com/dnceng/internal/_build/results?buildId=1034483 artifacts with our 2.1.25 release. The Microsoft.AspNetCore.App.nuspec file now contains the transitive references from dotnet/CoreFx and dotnet/core-setup.

In last `push`, I removed #30729 fixes that matter only between branding and baseline updates.

File: build/dependencies.props [hunk 1]
```diff
@@ -230,5 +230,10 @@
     <XunitExtensibilityExecutionPackageVersion>2.3.1</XunitExtensibilityExecutionPackageVersion>
     <XunitPackageVersion>2.4.0</XunitPackageVersion>
     <XunitRunnerVisualStudioPackageVersion>2.4.0</XunitRunnerVisualStudioPackageVersion>
+
+    <!-- Dependencies listed only to include otherwise-transitive references in Microsoft.AspNetCore.App.nuspec. -->
+    <MicrosoftDotNetPlatformAbstractionsPackageVersion>2.1.0</MicrosoftDotNetPlatformAbstractionsPackageVersion>
```
Comment (wtgodbe) [inline build/dependencies.props:235]: Are these all the latest?

**Review by wtgodbe — APPROVED** [review]
Seems reasonable

File: build/dependencies.props (hunk 1 above)
Comment (dougbu) [reply build/dependencies.props:235]: No. The two below are out-of-date. Should I check Microsoft.Extensions.DependencyModel and all of the `System.*` packages above❔

Comment (wtgodbe) [reply build/dependencies.props:235]: Yeah, it'd be better to have the latest for everything

File: build/dependencies.props [hunk 2]
```diff
@@ -132,11 +132,11 @@
     <MicrosoftIdentityModelClientsActiveDirectoryPackageVersion>3.14.2</MicrosoftIdentityModelClientsActiveDirectoryPackageVersion>
     <MicrosoftIdentityModelProtocolsOpenIdConnectPackageVersion>5.2.0</MicrosoftIdentityModelProtocolsOpenIdConnectPackageVersion>
     <MicrosoftIdentityModelProtocolsWsFederationPackageVersion>5.2.0</MicrosoftIdentityModelProtocolsWsFederationPackageVersion>
-    <MicrosoftNETCoreApp10PackageVersion>1.0.15</MicrosoftNETCoreApp10PackageVersion>
-    <MicrosoftNETCoreApp11PackageVersion>1.1.12</MicrosoftNETCoreApp11PackageVersion>
+    <MicrosoftNETCoreApp10PackageVersion>1.0.16</MicrosoftNETCoreApp10PackageVersion>
+    <MicrosoftNETCoreApp11PackageVersion>1.1.13</MicrosoftNETCoreApp11PackageVersion>
```
Comment (dougbu) [inline build/dependencies.props:136]: These two version variables are only mentioned in external-dependencies.props but I don't want to mess with deleting them

File: build/dependencies.props [hunk 3]
```diff
@@ -132,11 +132,11 @@
...
     <MicrosoftNETCoreApp20PackageVersion>2.0.9</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreWindowsApiSetsPackageVersion>1.0.1</MicrosoftNETCoreWindowsApiSetsPackageVersion>
-    <MicrosoftNETTestSdkPackageVersion>15.9.0</MicrosoftNETTestSdkPackageVersion>
+    <MicrosoftNETTestSdkPackageVersion>15.9.2</MicrosoftNETTestSdkPackageVersion>
```
Comment (dougbu) [inline build/dependencies.props:139]: This and `$(SystemNetHttpWinHttpHandlerPackageVersion)` may only matter in tests. They just got caught in my NuGet checkification

File: build/dependencies.props (hunk 1 above)
Comment (dougbu) [reply build/dependencies.props:235]: Agreed though I didn't check _everything_ in this file and instead focused on packages CoreFx, core-setup, extensions, and efcore

**Pull request merged**